### PR TITLE
POC for generating API endpoints using code generation, take 2

### PR DESCRIPTION
**Pull request by kpfleming**
This PR is a second proof-of-concept for generating API endpoints using the Jennifer code-generation module. It generates endpoints for Brotli compression and NGWAF as examples. For NGWAF, it includes configuration-during-enablement of the 'workspace_id', and configuration of 'workspace_id' and 'traffic_ramp' after enablement.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -46,11 +41,49 @@ dev-dependencies: ## Downloads the necessary dev dependencies.
...
+tidy: ## Cleans the Go module.
+	@echo "==> Tidying module"
+	@$(GO) mod tidy
+.PHONY: tidy
+
+generate: ## Builds and runs generators, to generate code for API endpoints and tests.
+	@echo "==> Removing existing generated code"
+	@rm -f fastly/products/*/api.go fastly/products/*/api_test.go
+	@echo "==> Building generators"
+	@$(GO) build -o generators ./internal/generators/...
```
Comment (ydnar) [inline Makefile:53]: Why build first? Can this just be go run?

Second, could these be embedded as go:generate lines?

File: fastly/client.go [hunk 2]
```diff
@@ -651,8 +651,8 @@ func checkResp(resp *http.Response, err error) (*http.Response, error) {
 	}
 }
 
-// decodeBodyMap is used to decode an HTTP response body into a mapstructure struct.
-func decodeBodyMap(body io.Reader, out any) error {
+// DecodeBodyMap is used to decode an HTTP response body into a mapstructure struct.
+func DecodeBodyMap(body io.Reader, out any) error {
```
Comment (ydnar) [inline fastly/client.go:655]: Can you have the exported func call the non-exported func, to minimize the size of this diff?

File: Makefile (hunk 1 above)
Comment (kpfleming) [reply Makefile:53]: Using `go run` in the `go:generate` lines would require that those lines have the relative path to the generator source code... that seemed a bit icky and fragile to me. This method produces binaries that are temporarily in the `PATH`, so the `go:generate` lines only need to include the name of the desired generator.

File: fastly/client.go (hunk 2 above)
Comment (kpfleming) [reply fastly/client.go:655]: I could... or I could put this change to expose `decodeBodyMap` into a separate PR.

Comment (kpfleming) [reply fastly/client.go:655]: This change (and other related changes) have now been moved to their own commit, which I'm maintaining in a local branch. Once this is ready, I'll push that set of changes in their own PR first.

**Comment by kpfleming** [comment]
Closing this as I've opted for a radically different approach which does not use code generation.

**Pull request closed without merging**